libresapi/src/api/IdentityHandler.cpp

```diff
@@ -282,6 +324,201 @@ void IdentityHandler::handleOwnIdsRequest(Request & /*req*/, Response &resp)
+	resp.mDataStream << makeKeyValue("gxs_id", data.mMeta.mGroupId.toStdString());
+
+	resp.mDataStream << makeKeyValue("pgp_id_known", data.mPgpKnown);
+	resp.mDataStream << makeKeyValue("pgp_id", data.mPgpId.toStdString());
+
+	std::string pgp_name;
+	if (data.mPgpKnown)
+	{
+		RsPeerDetails details;
+		rsPeers->getGPGDetails(data.mPgpId, details);
+		pgp_name = details.name;
+	}
+	else
+	{
+		if (data.mMeta.mGroupFlags & RSGXSID_GROUPFLAG_REALID_kept_for_compatibility)
+			pgp_name = "[Unknown node]";
+		else
+			pgp_name = "Anonymous Id";
+	}
+	resp.mDataStream << makeKeyValue("pgp_name", pgp_name);
+
+	time_t now = time(NULL);
+	resp.mDataStream << makeKeyValue("last_usage", difftime(now, data.mLastUsageTS));
+
+	bool isAnonymous = false;
+	if(!data.mPgpKnown)
+	{
+		if (!(data.mMeta.mGroupFlags & RSGXSID_GROUPFLAG_REALID_kept_for_compatibility))
+			isAnonymous = true;
+	}
+	resp.mDataStream << makeKeyValue("anonymous", isAnonymous);
+
+
+	bool isOwnId = (data.mMeta.mSubscribeFlags & GXS_SERV::GROUP_SUBSCRIBE_ADMIN);
+	resp.mDataStream << makeKeyValue("own", isOwnId);
+
+	std::string type;
+	if(isOwnId)
+	{
+		if (data.mPgpKnown && !data.mPgpId.isNull())
+			type = "Identity owned by you, linked to your Retroshare node";
+		else
+			type = "Anonymous identity, owned by you";
+	}
+	else if (data.mMeta.mGroupFlags & RSGXSID_GROUPFLAG_REALID_kept_for_compatibility)
+	{
+		if (data.mPgpKnown)
+		{
+			if (rsPeers->isGPGAccepted(data.mPgpId))
+				type = "Linked to a friend Retroshare node";
+			else
+				type = "Linked to a known Retroshare node";
+		}
+		else
+			type = "Linked to unknown Retroshare node";
+	}
+	else
+		type = "Anonymous identity";
+
+	resp.mDataStream << makeKeyValue("type", type);
+
+	resp.mDataStream << makeKeyValue("bannned_node", rsReputations->isNodeBanned(data.mPgpId));
+
+	RsReputations::ReputationInfo info;
+	rsReputations->getReputationInfo(RsGxsId(data.mMeta.mGroupId), data.mPgpId, info);
+	resp.mDataStream << makeKeyValue("friends_positive_votes", info.mFriendsPositiveVotes);
+	resp.mDataStream << makeKeyValue("friends_negative_votes", info.mFriendsNegativeVotes);
+	resp.mDataStream << makeKeyValue("overall_reputation_level", (int)info.mOverallReputationLevel);
+	resp.mDataStream << makeKeyValue("own_opinion", (int)info.mOwnOpinion);
+
+	RsIdentityDetails details;
+	mRsIdentity->getIdDetails(RsGxsId(data.mMeta.mGroupId), details);
+	StreamBase& usagesStream = resp.mDataStream.getStreamToMember("usages");
+	usagesStream.getStreamToMember();
+
+	for(std::map<RsIdentityUsage,time_t>::const_iterator it(details.mUseCases.begin()); it != details.mUseCases.end(); ++it)
+	{
+		usagesStream.getStreamToMember() << makeKeyValue("usage_time", difftime(now, data.mLastUsageTS));
+		usagesStream.getStreamToMember() << makeKeyValue("usage_service", (int)(it->first.mServiceId));
+		usagesStream.getStreamToMember() << makeKeyValue("usage_case", (int)(it->first.mUsageCode));

```

Comment:
every call to getStreamToMember() will make a new object. It would be more logical to have one object in the array for each use case. Stream operator << returns a reference to the stream, so you can just chain the values.

    usagesStream.getStreamToMember()
        << makeKeyValue("usage_time", difftime(now, data.mLastUsageTS))
        << makeKeyValue("usage_service", (int)(it->first.mServiceId))
        << makeKeyValue("usage_case", (int)(it->first.mUsageCode))
    ;